From linux-bluetooth-owner@vger.kernel.org Wed Sep 16 08:35:32 2009
Return-Path: <linux-bluetooth-owner@vger.kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.2.5 (2008-06-10) on
 bungle.evilgeniuses.org.uk
X-Spam-Level: 
X-Spam-Status: No, score=-3.7 required=5.0 tests=AWL,BAYES_00,
 RCVD_IN_DNSWL_MED autolearn=ham version=3.2.5
X-Original-To: hadess@hadess.net
Delivered-To: hadess@bungle.evilgeniuses.org.uk
Received: from vger.kernel.org (vger.kernel.org [209.132.176.167]) by
 bungle.evilgeniuses.org.uk (Postfix) with ESMTP id BC0161C3BCC0 for
 <hadess@hadess.net>; Wed, 16 Sep 2009 08:35:32 +0100 (BST)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand id
 S1751251AbZIPHf1 (ORCPT <rfc822;hadess@hadess.net>); Wed, 16 Sep 2009
 03:35:27 -0400
Received: (majordomo@vger.kernel.org) by vger.kernel.org id
 S1751490AbZIPHf1 (ORCPT <rfc822;linux-bluetooth-outgoing>); Wed, 16 Sep
 2009 03:35:27 -0400
Received: from mga01.intel.com ([192.55.52.88]:35513 "EHLO mga01.intel.com"
 rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP id S1751251AbZIPHf0
 (ORCPT <rfc822;linux-bluetooth@vger.kernel.org>); Wed, 16 Sep 2009 03:35:26
 -0400
Received: from fmsmga002.fm.intel.com ([10.253.24.26]) by
 fmsmga101.fm.intel.com with ESMTP; 16 Sep 2009 00:31:16 -0700
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="4.44,396,1249282800"; 
 d="scan'208,223";a="493920533"
Received: from unknown (HELO yangyi-dev.bj.intel.com) ([172.16.180.135]) by
 fmsmga002.fm.intel.com with ESMTP; 16 Sep 2009 00:27:42 -0700
Date:	Wed, 16 Sep 2009 15:33:59 +0800
From:	Yi Yang <yi.y.yang@intel.com>
To:	linux-bluetooth@vger.kernel.org
Subject: [PATCH resent] bluez: correct udev rule for Dell mouse
Message-ID: <20090916073359.GA22265@yangyi-dev.bj.intel.com>
MIME-Version: 1.0
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline
User-Agent: Mutt/1.5.18 (2008-05-17)
Sender:	linux-bluetooth-owner@vger.kernel.org
Precedence: bulk
List-ID: <linux-bluetooth.vger.kernel.org>
X-Mailing-List:	linux-bluetooth@vger.kernel.org
X-Evolution-Source: imap://hadess@bungle.evilgeniuses.org.uk/
Content-Transfer-Encoding: 8bit

>From dbcfcf5dfbdef2377f64c7cd737886e31ae17615 Mon Sep 17 00:00:00 2001
From: Yi Yang <yi.y.yang@intel.com>
Date: Wed, 16 Sep 2009 15:19:13 +0800
Subject: [PATCH] Correct udev rule for Dell mouse

Note: this rule should just for Dell bluetooth mouse, but it
does function for Dell general USB mouse, such as this one:

    ID 413c:3010 Dell Computer Corp. Optical Wheel Mouse

We should let it just function for Dell bluetooth mouse, udev
144 and later has included rules for hid2hci
extras/hid2hci/70-hid2hci.rules which has correct one, so we
should use that one
---
 scripts/bluetooth-hid2hci.rules |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/scripts/bluetooth-hid2hci.rules b/scripts/bluetooth-hid2hci.rules
index 1b231d1..eb63292 100644
--- a/scripts/bluetooth-hid2hci.rules
+++ b/scripts/bluetooth-hid2hci.rules
@@ -8,7 +8,8 @@
 #   413c:8154
 #   413c:8158
 #   413c:8162
-ACTION=="add", ENV{ID_VENDOR}=="413c", ENV{ID_CLASS}=="mouse", ATTRS{bmAttributes}=="e0", KERNEL=="mouse*", RUN+="/usr/sbin/hid2hci --method dell -v $env{ID_VENDOR} -p $env{ID_MODEL} --mode hci"
+ATTR{bInterfaceClass}=="03", ATTR{bInterfaceSubClass}=="01", ATTR{bInterfaceProtocol}=="02", ATTRS{bDeviceClass}=="00", ATTRS{idVendor}=="413c", ATTRS{bmAttributes}=="e0", \
+    RUN+="hid2hci --method dell -v $attr{idVendor} -p $attr{idProduct} --mode hci"
 
 # Logitech devices
 ACTION=="add", ENV{ID_VENDOR}=="046d", ENV{ID_MODEL}=="c703" RUN+="/usr/sbin/hid2hci --method logitech -v $env{ID_VENDOR} -p $env{ID_MODEL} --mode hci"
-- 
1.6.0.6

--
To unsubscribe from this list: send the line "unsubscribe linux-bluetooth" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

